server/statistics/hot_cache.go

```diff
@@ -376,124 +67,57 @@ func (w *HotSpotCache) Update(item *HotSpotPeerStat) {
 }
 
 // RegionStats returns hot items according to kind
-func (w *HotSpotCache) RegionStats(kind FlowKind) map[uint64][]*HotSpotPeerStat {
-	var flowMap map[uint64]cache.Cache
+func (w *HotSpotCache) RegionStats(kind FlowKind) map[uint64][]*HotPeerStat {
+	var peersOfStore map[uint64]cache.Cache
 	switch kind {
 	case WriteFlow:
-		flowMap = w.writeFlow.hotStoreStats
+		peersOfStore = w.writeFlow.peersOfStore
 	case ReadFlow:
-		flowMap = w.readFlow.hotStoreStats
+		peersOfStore = w.readFlow.peersOfStore
 	}
-	res := make(map[uint64][]*HotSpotPeerStat)
-	for storeID, elements := range flowMap {
-		values := elements.Elems()
-		stat, ok := res[storeID]
-		if !ok {
-			stat = make([]*HotSpotPeerStat, len(values))
-			res[storeID] = stat
-		}
+
+	res := make(map[uint64][]*HotPeerStat)
+	for storeID, peers := range peersOfStore {
+		values := peers.Elems()
+		stat := make([]*HotPeerStat, len(values))
+		res[storeID] = stat
 		for i := range values {
-			stat[i] = values[i].Value.(*HotSpotPeerStat)
+			stat[i] = values[i].Value.(*HotPeerStat)
 		}
 	}
 	return res
 }
 
 // RandHotRegionFromStore random picks a hot region in specify store.
-func (w *HotSpotCache) RandHotRegionFromStore(storeID uint64, kind FlowKind, hotThreshold int) *HotSpotPeerStat {
-	stats, ok := w.RegionStats(kind)[storeID]
-	if !ok {
-		return nil
-	}
-	for _, i := range rand.Perm(len(stats)) {
-		if stats[i].HotDegree >= hotThreshold {
-			return stats[i]
+func (w *HotSpotCache) RandHotRegionFromStore(storeID uint64, kind FlowKind, hotDegree int) *HotPeerStat {
```

Comment:
This function is refactored using another coding style, I think both are OK to me.

The original style uses a return to avoid nesting too much.